docs/adr/0005-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,83 @@
+
+## Decision Drivers
+
+* We want to get changes merged and released to our consumer applications faster.
+* We want to simplify pushing new code to applications, especially urgent bug fixes.
+* We want to move towards the future monorepo world.
+* Mozilla is moving towards rapid releases for mobile code.
+* We are willing to accept the overhead of upgrading application-services more frequently in application repositories.
+
+## Considered Options
+
+* **(A) Keep our current release process**
+    * Keep everything the same for a while and wait for other code to move into the monorepo.
+* **(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+    * Phase 1: update our branching model and release process
+        * Update our branches to match how moz-central: we would have a `main`, `beta` and `release` branch, with nightly being built off of main, as well as `esr-*` branches for older ESR releases.
+        * Automate a nightly release of our `main` branch.
+          * Create a commit that updates our version number to something like `[firefox-version].0.nightly-[YYYYMMDD]`, similar to what `prepare-release.py` does now.
+            * Make sure that the github CI can push this commit without needing PR approval.
+            * The version string is composed of major, minor, and patch components.  This matches firefox-android, but not moz-central, which skips the minor component.  The reason to have a minor component is to support weekly releases for mobile apps.  They may choose to include new features on those apps, which may require us to uplift new features to our release version and that would be a minor version bump.
+          * Add a git tag for the commit based on that version number (`v[firefox-version].0.nightly-[YYMMDD]`).
+          * Run the `github-release` CI to publish our code to our consumers (publish maven packages, start the `rust-components-swift` CI, etc).
+        * Automate releases from the `beta`, `release`, and `esr-*` branches whenever a PR is merged
+          * Create a commit that updates our version number to `[firefox-version].X.[Y+1]`
+          * Add a git tag for the commit based on that version number (`v[firefox-version].X.[Y+1])
+          * Run the `github-release` CI to publish our code to our consumers (publish maven packages, start the `rust-components-swift` CI, etc).
+          * Maybe we should create PRs to merge the changes back forward to other branches.  For example, merging changes from `release` into `beta` and `beta` into `main`.
```

Comment:
I'd love to get some wisdom from releng for this.  How does this work for moz-central?